To: vim_dev@googlegroups.com
Subject: Patch 7.3.464
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.464
Problem:    Compiler warning for sprintf.
Solution:   Put the length in a variable. (Dominique Pelle)
Files:	    src/version.c


*** ../vim-7.3.463/src/version.c	2012-03-07 14:57:50.000000000 +0100
--- src/version.c	2012-03-07 17:58:41.000000000 +0100
***************
*** 2187,2198 ****
  	    /* Check for 9.9x or 9.9xx, alpha/beta version */
  	    if (isalpha((int)vers[3]))
  	    {
! 		if (isalpha((int)vers[4]))
! 		    sprintf((char *)vers + 5, ".%d%s", highest_patch(),
! 							   mediumVersion + 5);
! 		else
! 		    sprintf((char *)vers + 4, ".%d%s", highest_patch(),
! 							   mediumVersion + 4);
  	    }
  	    else
  		sprintf((char *)vers + 3, ".%d", highest_patch());
--- 2189,2197 ----
  	    /* Check for 9.9x or 9.9xx, alpha/beta version */
  	    if (isalpha((int)vers[3]))
  	    {
! 		int len = (isalpha((int)vers[4])) ? 5 : 4;
! 		sprintf((char *)vers + len, ".%d%s", highest_patch(),
! 							 mediumVersion + len);
  	    }
  	    else
  		sprintf((char *)vers + 3, ".%d", highest_patch());
*** ../vim-7.3.463/src/version.c	2012-03-07 14:57:50.000000000 +0100
--- src/version.c	2012-03-07 17:58:41.000000000 +0100
***************
*** 716,717 ****
--- 716,719 ----
  {   /* Add new patch number below this line */
+ /**/
+     464,
  /**/

-- 
hundred-and-one symptoms of being an internet addict:
152. You find yourself falling for someone you've never seen or hardly
     know, but, boy can he/she TYPE!!!!!!

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
